nestkernel/node_collection.cpp

```diff
@@ -240,7 +240,13 @@ NodeCollectionPrimitive::NodeCollectionPrimitive( index first,
   , last_( last )
   , model_id_( model_id )
   , metadata_( meta )
+  , nodes_have_no_proxies_( not kernel().model_manager.get_model( model_id_ )->has_proxies() )
 {
+  if ( not all_model_ids_same_as_expected_( model_id_ ) )
+  {
+    throw BadProperty( "model ids does not match" );
```

Comment:
```suggestion
    throw BadProperty( "All nodes must have the same model model id" );
```
Maybe improve further with Compose to inform where it goes wrong. Even better: change name to `assert_consistent_model_ids_()` and throw exceptions directly in there.